backend/app/verification.py

```diff
@@ -884,6 +912,78 @@ def unverify(
         worker.republish_app.send(app_id)
 
 
+@router.post("/{app_id}/enroll", status_code=204, tags=["verification"])
+def enroll(
+    login=Depends(logged_in),
+    app_id: str = Path(
+        min_length=6,
+        max_length=255,
+        pattern=r"^[A-Za-z_][\w\-\.]+$",
+        examples=["org.gnome.Glade"],
+    ),
+):
+    is_direct_upload = models.DirectUploadApp.by_app_id(sqldb, app_id) is not None
+
+    if is_direct_upload:
+        return
+
+    is_verified = _get_existing_verification(app_id)
+    if is_verified and not is_direct_upload:
+        _create_direct_upload_app(login.user, app_id)
+        _archive_github_repo(app_id)
+
+
+@router.post("/{app_id}/archive", status_code=204, tags=["verification"])
+def archive(
+    request: ArchiveRequest,
+    login=Depends(logged_in),
+    app_id: str = Path(
+        min_length=6,
+        max_length=255,
+        pattern=r"^[A-Za-z_][\w\-\.]+$",
+        examples=["org.gnome.Glade"],
+    ),
+):
+    if not config.settings.flat_manager_api:
+        return
+
+    direct_upload_app = models.DirectUploadApp.by_app_id(sqldb, app_id)
+    if direct_upload_app is not None:
+        if direct_upload_app.archived:
+            return
+
+        direct_upload_app.archived = True
+        sqldb.session.commit()
+
+    if not direct_upload_app:
```

Comment:
Kind of confused, why this is here and not the same as 951 - but it's late :)

Reply:
Direct upload vs GitHub app, but I guess I expressed conditions in a confusing way. Changed it